#! /bin/sh /usr/share/dpatch/dpatch-run
## 11_SetFigFont-params.dpatch by Roland Rosenfeld <roland@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Rename TeX macro \SetFigFont to \SetFigFontNFSS if NFSS ist defined.
## DP: this should work around problems of users using PSTEX files generated
## DP: by old and by new fig2dev in the same document.

@DPATCH@
diff -urNad transfig~/fig2dev/dev/genepic.c transfig/fig2dev/dev/genepic.c
--- transfig~/fig2dev/dev/genepic.c
+++ transfig/fig2dev/dev/genepic.c
@@ -1227,10 +1227,10 @@
 
 #ifdef NFSS
     if ( FontSizeOnly )
-	fprintf(tfp, "{\\SetFigFont{%d}{%.1f}",
+	fprintf(tfp, "{\\SetFigFontNFSS{%d}{%.1f}",
 		texsize, baselineskip );
     else
-	fprintf(tfp, "{\\SetFigFont{%d}{%.1f}{%s}{%s}{%s}",
+	fprintf(tfp, "{\\SetFigFontNFSS{%d}{%.1f}{%s}{%s}{%s}",
 		texsize, baselineskip,
 		TEXFAMILY(text->font),TEXSERIES(text->font),
 		TEXSHAPE(text->font));
diff -urNad transfig~/fig2dev/dev/genmp.c transfig/fig2dev/dev/genmp.c
--- transfig~/fig2dev/dev/genmp.c
+++ transfig/fig2dev/dev/genmp.c
@@ -1273,7 +1273,7 @@
     /* not default font: set the font and font size. */
     if (! isdefaultfont (t->flags, t->font)) {
 #ifdef NFSS
-	fprintf(tfp,"       \\SetFigFont{%d}{%.1f}{%s}{%s}{%s}%%\n",
+	fprintf(tfp,"       \\SetFigFontNFSS{%d}{%.1f}{%s}{%s}{%s}%%\n",
 		texsize, baselineskip,
 		TEXFAMILY(t->font),TEXSERIES(t->font),TEXSHAPE(t->font));
 #else
diff -urNad transfig~/fig2dev/dev/genpictex.c transfig/fig2dev/dev/genpictex.c
--- transfig~/fig2dev/dev/genpictex.c
+++ transfig/fig2dev/dev/genpictex.c
@@ -637,7 +637,7 @@
 	  baselineskip = (texsize * 1.2);
 
 #ifdef NFSS
- 	  fprintf(tfp, "\\put{\\SetFigFont{%d}{%.1f}{%s}{%s}{%s}",
+ 	  fprintf(tfp, "\\put{\\SetFigFontNFSS{%d}{%.1f}{%s}{%s}{%s}",
 				 texsize, baselineskip,
 				 TEXFAMILY(t->font),TEXSERIES(t->font),TEXSHAPE(t->font));
 #else
diff -urNad transfig~/fig2dev/dev/setfigfont.c transfig/fig2dev/dev/setfigfont.c
--- transfig~/fig2dev/dev/setfigfont.c
+++ transfig/fig2dev/dev/setfigfont.c
@@ -36,15 +36,15 @@
 
     if ( FontSizeOnly )
     	fprintf(tfp, "%%\n\
-\\begingroup\\makeatletter\\ifx\\SetFigFont\\undefined%%\n\
-\\gdef\\SetFigFont#1#2{%%\n\
+\\begingroup\\makeatletter\\ifx\\SetFigFontNFSS\\undefined%%\n\
+\\gdef\\SetFigFontNFSS#1#2{%%\n\
   \\fontsize{#1}{#2pt}%%\n\
   \\selectfont}%%\n\
 \\fi\\endgroup%%\n");
     else
 	fprintf(tfp, "%%\n\
-\\begingroup\\makeatletter\\ifx\\SetFigFont\\undefined%%\n\
-\\gdef\\SetFigFont#1#2#3#4#5{%%\n\
+\\begingroup\\makeatletter\\ifx\\SetFigFontNFSS\\undefined%%\n\
+\\gdef\\SetFigFontNFSS#1#2#3#4#5{%%\n\
   \\reset@font\\fontsize{#1}{#2pt}%%\n\
   \\fontfamily{#3}\\fontseries{#4}\\fontshape{#5}%%\n\
   \\selectfont}%%\n\
